### DISCUSSION TITLE
Support for MySQL 8 in version 2.3.3

### DISCUSSION THREAD

**Discussion by gabyvlad96** [sidorares/node-mysql2#2937] [Q&A]
Hello, I would need to know if version 2.3.3 is compatible with MySQL >=8.0
There isn't much info about compatibility as far as I researched.
Thanks!

**Comment by sidorares** [comment]
which part of mysql 8 functionality is important to you? At the protocol level there is almost no difference. I believe 2.3.3 does not support `caching_sha2_password` auth plugin, if you have an option to configure the server to use `mysql_native_password` plugin you should be able to connect and communicate with > 8.0 server without problems ( see for example https://www.stackhero.io/en/services/MySQL/documentations/Troubleshooting/How-to-solve-MySQL-error-Authentication-plugin-cachingsha2password-cannot-be-loaded on how to configure auth plugin for the server )

**Comment by gabyvlad96** [comment]
Hi, let me provide more context.
We are using mysql2 only as a dependency for Sequelize. We plan to upgrade from MySQL 5.7 to MySQL 8 so as part of this I want to make sure both Sequelize 6 and mysql2 2.33 we use are compatible.
Sequelize [confirms here](https://sequelize.org/releases/#mysql-support-table) that it supports MySQL 8 with our current versions. From this I understand that implicitly 2.3.3 would also work but I just need to be 100% sure hence the question.

I will look into the `caching_sha2_password` thanks for providing that.
As for functionality, I don't have all the info so I need to check on it. But I don't believe there's a specific functionality we need.
